ENH : user register 생성

hun1398 opened the pull request:
- gradle 설정
- properties 설정 mysql 설정, mybatis 설정

hun1398 committed:
ENH : user register 생성
- gradle 설정
- properties 설정
mysql 설정, mybatis 설정

j-lab-dev commented on IdolTicketing/src/main/java/com/IdolTicketing/IdolTicketingApplication.java:
```diff
@@ -0,0 +1,13 @@
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+
+@SpringBootApplication
```
@SpringBootApplication 어노테이션이 실행될때 상세 과정을 주석으로 추가해주세요.

hun1398 replied:
@j-lab-dev  주석으로 추가했습니다. 확인 부탁드립니다.

ccommit-dev replied:
@hun1398 @SpringBootApplication 다음부턴 리뷰 받은 브랜치에 추가 부탁드립니다

j-lab-dev commented on IdolTicketing/src/main/java/com/IdolTicketing/controller/UserController.java:
```diff
@@ -0,0 +1,33 @@
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+@RestController
```
@RestController 어노테이션의 기능을 최대한 상세히 주석으로 추가해주세요.

hun1398 replied:
@j-lab-dev 주석으로 추가했습니다. 확인 부탁드립니다.

j-lab-dev replied:
분산 시스템을 왜사용해야 할까요?
댓글로 작성해주세요.

hun1398 replied:
분산 시스템에서 어느 요소의 일부가 고장 나더라도 전체의 시스템은 계속 동작할 수 있도록 하기 위해서 입니다.

ccommit-dev replied:
추가적인 이유는 어떤게 있을까요?

j-lab-dev commented on IdolTicketing/src/main/java/com/IdolTicketing/controller/UserController.java:
```diff
@@ -0,0 +1,33 @@
+import org.springframework.web.bind.annotation.*;
+
+@RestController
+@RequestMapping(value = "/users")
```
REST API의 정의와 규칙에 대해 댓글 또는 주석으로 추가해주세요.

hun1398 replied:
@j-lab-dev 
REST API의 정의
-REST는 HTTP 프로토콜을 통해 API를 설계하기 위한 아키텍처 스타일입니다. 
-REST의 함수를 계층화하는 기능은 유연성을 더욱 향상시킵니다.
 (EX.MVC 패턴)

j-lab-dev replied:
REST API와 MVC 패턴의 어떤 상관관계 있을까요?

hun1398 replied:
MVC 패턴으로 만든 REST 아키텍처를 준수한 API에서 데이터를 입출력합니다.

ccommit-dev replied:
REST 아키텍처를 준수하려면 어떤 내용을 지켜야 할까요?

hun1398 replied:
소문자 사용, 하이픈 사용, 마지막에는 슬래시 사용 않기,명사 사용,확장자 표지 않기,복수형 사용,계층 관계 슬래시로 구분 등 있습니다.
ex) i-dol/users/books/list

j-lab-dev commented on IdolTicketing/src/main/java/com/IdolTicketing/controller/UserController.java:
```diff
@@ -0,0 +1,33 @@
+public class UserController {
+
+    @Autowired
+    UserMapper userMapper;
```
Service 객체 안에서 userMapper 클래스를 주입받게 바꿔주세요.
이유는 비즈니스 로직은 서비스 객체에서 처리해야 추가 요구사항 개발시에 기존코드가 변경되지 않습니다. 
(객체지향 원칙, SRP 원칙 위배 사항)

hun1398 replied:
@j-lab-dev 반영하겠습니다.

j-lab-dev commented on IdolTicketing/src/main/java/com/IdolTicketing/controller/UserController.java:
```diff
@@ -0,0 +1,33 @@
+    @Autowired
+    UserMapper userMapper;
+
+    private final UserService userService;
```
final 키워드를 사용하신 이유를 댓글로 달아주세요.

hun1398 replied:
@j-lab-dev  캡슐화,SRP,
final 키워드는 단어 그대로 마지막 이라는 뜻으로 다른 클래스가 상속 할 수 없게 됩니다.

j-lab-dev commented on IdolTicketing/src/main/resources/Mapper/UserMapper.xml:
```diff
@@ -0,0 +1,11 @@
+        (userId,name,password,email,phone,address,isAdmin)
+        VALUES (#{userId},#{name},#{password},#{email},#{phone},
+        #{address},#{isAdmin})
+    </insert>
```
user Controller 이슈에서 생성 이외에 수정 검색 내용도 추가해주세요.

hun1398 replied:
@j-lab-dev 추가했습니다.

ccommit-dev reviewed (commented):
@hun1398  코드리뷰 해당부분 확인 및 반영 부탁드립니다.

hun1398 commented on .gitignore:
```diff
@@ -0,0 +1 @@
+/.idea/
```
@j-lab-dev 멘토님 답변드립니다.

- 보안상으로 위험성이 있는 파일
- 프로젝트와 관계없는 파일
- 용량이 너무 커서 제외해야되는 파일

위 3가지와 연관된 내용은 아래 내용으로 생각합니다.
target
out
build
.gradle

확인 부탁드립니다.

hun1398 committed:
ENH : userController
- login,update,logout,delete 추가 수정
gradle
- AOP 설정
userMapper 수정
userService 수정
userDTO 추가 수정 및 생성
Login Check AOP 생성
enum role 생성
SessionUtil 생성

j-lab-dev commented on IdolTicketing/src/main/java/com/IdolTicketing/aop/AuthCheckAspect.java:
```diff
@@ -0,0 +1,71 @@
+                break;
+            }
+        }
+        if( Collections.list(((StandardSessionFacade) session).getAttributeNames()).size() >0){
```
session 에 사이즈가 0보다 큰상황인데 
if (sessionUserId == null) 을 또 비교할 필요가 있을까요?

hun1398 replied:
수정했습니다.

j-lab-dev commented on IdolTicketing/src/main/java/com/IdolTicketing/aop/AuthCheckAspect.java:
```diff
@@ -0,0 +1,71 @@
+
+        if (jp.getArgs() != null){
+            signatureArgs[0] = sessionUserId;
+            signatureArgs[1] = isAdmin;
```
isAdmin 값은 전체 Controller에서 사용하는 값일까요?
아니라면 굳이 파라미터로 넘길 필요는 없어보입니다. 고민해보시고 댓글로 의견 부탁드립니다.

hun1398 replied:
전체 컨트롤러에서 사용하는 값입니다.

j-lab-dev commented on IdolTicketing/src/main/java/com/IdolTicketing/controller/UserController.java:
```diff
@@ -27,7 +34,91 @@ public UserController(UserService userService) {
+        } else
+            SessionUtil.setLoginAdminId(session, userInfo.getUserId());
+        return new ResponseEntity<>(UserResponseDTO.builder()
+                .code(202)
```
202의 뜻은 어떤값일까요?

j-lab-dev commented on IdolTicketing/src/main/java/com/IdolTicketing/controller/UserController.java:
```diff
@@ -27,7 +34,91 @@ public UserController(UserService userService) {
+                .massage("ADMIN").build(), HttpStatus.OK);
+    }
+
+    @PatchMapping("/update")
```
PUT VS Patch 차이를 댓글로 작성해주세요

hun1398 replied:
put의 리소스의 전체를 업데이트하며, patch는 일부분만 변경합니다.

j-lab-dev commented on IdolTicketing/src/main/java/com/IdolTicketing/dto/UserDTO.java:
```diff
@@ -22,7 +22,7 @@ public UserDTO(){
     }
 
-    public UserDTO(String userId,String name,String password, String email,String phone,String address,boolean isAdmin,
+    public UserDTO(String userId,String name,String password, String email,int phone,String address,boolean isAdmin,
```
phon은 왜 int로 변경하셨나요?

hun1398 replied:
phone을 null로 안받기 위해서 입니다.

ccommit-dev reviewed (commented):
리뷰 내용 재확인후 답변 또는 다시 수정 부탁드립니다.

ccommit-dev commented:
해당 브랜치는 머지 부탁드립니다.
